pysmt/solvers/cvc4.py

```diff
@@ -288,9 +288,16 @@ def walk_equals(self, formula, args, **kwargs):
         return self.mkExpr(CVC4.EQUAL, args[0], args[1])
 
     def walk_times(self, formula, args, **kwargs):
-        if not args[0].isConst() and not args[1].isConst():
-            raise NonLinearError(formula)
-        return self.mkExpr(CVC4.MULT, args[0], args[1])
+        res = args[0]
+        nl_count = 0 if res.isConst() else 1
```

Comment:
Looks like isConst() returns whether you have a constant term and not a constant expression.

https://travis-ci.org/pysmt/pysmt/jobs/153900213